terraso_backend/apps/soil_id/models/depth_dependent_soil_data.py

```diff
@@ -13,13 +13,298 @@
 # You should have received a copy of the GNU Affero General Public License
 # along with this program. If not, see https://www.gnu.org/licenses/.
 
+from django.core.validators import MaxValueValidator, MinValueValidator
 from django.db import models
+from django.utils.translation import gettext_lazy as _
 
 from apps.core.models.commons import BaseModel
 from apps.soil_id.models.soil_data import SoilData
 
+OTHER = "other"
+
 
 class DepthDependentSoilData(BaseModel):
     soil_data_input = models.ForeignKey(SoilData, on_delete=models.CASCADE)
     depth_top = models.IntegerField(blank=True)
     depth_bottom = models.IntegerField(blank=True)
+
+    SAND = "sand"
+    LOAMY_SAND = "loamy sand"
+    SANDY_LOAM = "sandy loam"
+    SILT_LOAM = "slit loam"
+    SILT = "silt"
+    LOAM = "loam"
+    SANDY_CLAY_LOAM = "sandy clay loam"
+    SILTY_CLAY_LOAM = "silty clay loam"
+    CLAY_LOAM = "clay loam"
+    SANDY_CLAY = "sandy clay"
+    SILTY_CLAY = "silty clay"
+    CLAY = "clay"
+
+    TEXTURES = (
+        (SAND, _("Sand")),
+        (LOAMY_SAND, _("Loamy sand")),
+        (SANDY_LOAM, _("Sandy loam")),
+        (SILT_LOAM, _("Silt loam")),
+        (SILT, _("Silt")),
+        (LOAM, _("Loam")),
+        (SANDY_CLAY_LOAM, _("Sandy clay loam")),
+        (SILTY_CLAY_LOAM, _("Silty clay loam")),
+        (CLAY_LOAM, _("Clay loam")),
+        (SANDY_CLAY, _("Sandy clay")),
+        (SILTY_CLAY, _("Silty clay")),
+        (CLAY, _("Clay")),
+    )
+    texture = models.CharField(null=True, choices=TEXTURES)
+
+    ROCK_FRAGMENT_0_1 = "0 - 1%"
+    ROCK_FRAGMENT_1_15 = "1 - 15%"
+    ROCK_FRAGMENT_15_35 = "15 - 35%"
+    ROCK_FRAGMENT_35_60 = "35 - 60%"
+    ROCK_FRAGMENT_60 = "> 60%"
+
+    ROCK_FRAGMENT_VOLUMES = (
+        (ROCK_FRAGMENT_0_1, _("0 - 1%")),
+        (ROCK_FRAGMENT_1_15, _("1 - 15%")),
+        (ROCK_FRAGMENT_15_35, _("15 - 35%")),
+        (ROCK_FRAGMENT_35_60, _("35 - 60%")),
+        (ROCK_FRAGMENT_60, _("> 60%")),
+    )
+    rock_fragment_volume = models.CharField(null=True, choices=ROCK_FRAGMENT_VOLUMES)
+
+    SUBSTEP_2_5 = "2.5"
+    SUBSTEP_5 = "5"
+    SUBSTEP_7_5 = "7.5"
+    SUBSTEP_10 = "10"
+
+    COLOR_HUES_SUBSTEPS = (
+        (SUBSTEP_2_5, _("2.5")),
+        (SUBSTEP_2_5, _("5")),
+        (SUBSTEP_7_5, _("7.5")),
+        (SUBSTEP_10, _("10")),
+    )
+    color_hue_substeps = models.CharField(null=True, choices=COLOR_HUES_SUBSTEPS)
+
+    RED = "R"
+    YELLOW_RED = "YR"
+    YELLOW = "Y"
+    GREEN_YELLOW = "GY"
+    GREEN = "G"
+    BLUE = "B"
+    BLUE_GREEN = "BG"
```

Comment:
question: what is the reason for abbreviating these values? Is that part of a standard notation?

Reply:
@ltseng What are your thoughts on this? I just took what was in the spec doc.

Reply:
Yes, this is part of the standard notation.

Reply:
These are part of the standard notation. https://munsell.com/about-munsell-color/how-color-notation-works/munsell-hue-circle-comparison/